test/oncluster/scenario_github_test.go

```diff
@@ -32,10 +33,14 @@ Notes:
 
 func resolveGitVars() (gitRepoUrl string, gitRef string) {
 	// On a GitHub Action (Pull Request) these variables will be set
-	// https://docs.github.com/en/actions/learn-github-actions/variables
+	// https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables
 	gitRepo := common.GetOsEnvOrDefault("GITHUB_REPOSITORY", "knative/func")
-	gitRef = common.GetOsEnvOrDefault("GITHUB_REF", "main")
 	gitRepoUrl = "https://github.com/" + gitRepo + ".git"
+
+	gitRef = common.GetOsEnvOrDefault("GITHUB_REF", "main")
```

Comment:
Doesn't `GITHUB_HEAD_REF` contain what you want?

Reply:
yea.. I'll gonna give a try

Reply:
hmm.. `GITHUB_HEAD_REF` does not work here in this case.. it returns the branch name.. https://github.com/knative/func/actions/runs/5812963695/job/15759506289?pr=1917#step:10:182 .. so just rolled back to the first implementation.